lib/facter/resolvers/partitions.rb

```diff
@@ -121,7 +122,18 @@ def execute_and_extract_blkid_info
         def populate_from_lsblk(partition_name, blkid_and_lsblk)
           return {} unless available?('lsblk', blkid_and_lsblk)
 
-          blkid_and_lsblk[:lsblk] ||= Facter::Core::Execution.execute('lsblk -fp', logger: log)
+          lsblk_version_raw = Facter::Core::Execution.execute('lsblk --version 2>&1', logger: log)
+          lsblk_version = lsblk_version_raw.match(/ \d\.\d+/)[0].to_f
+
+          # The -p or --paths option was added in lsblk 2.23, anything before that it makes more sense to return and fall back to blkid
+          return {} if lsblk_version < 2.23
+
+          # lsblk 2.25 added support for GPT partition type GUIDs
+          blkid_and_lsblk[:lsblk] ||= if lsblk_version >= 2.25
```

Comment:
Might it be easier to run `lsblk` with the longer args and `rescue` if it errors out?